### PR TITLE
Don't pull OSTree ref if it is already in repo

### PR DESCRIPTION
**Pull request by OYTIS**
Should help preventing trying to pull an image every 10 seconds.

### CODE REVIEW

File: src/ostree.cc [hunk 1]
```diff
@@ -55,6 +55,29 @@ data::InstallOutcome OstreeManager::pull(const Config &config, const KeyManager
   return data::InstallOutcome(data::OK, "Pulling ostree image was successful");
 }
 
+data::InstallOutcome OstreeManager::find_or_pull(const Config &config, const KeyManager &keys, const std::string &refhash) {
+  OstreeRepo *repo = NULL;
+  const char *const commit_ids[] = {refhash.c_str()};
+  GCancellable *cancellable = NULL;
+  GError *error = NULL;
+  GVariantBuilder builder;
+  GVariant *options;
```
Comment (patrickvacek) [inline src/ostree.cc]: `commit_ids`, `builder`, and `options` are not actually used here. But does it make sense to merge this into the `pull` function anyway? Normally I wouldn't advocate for making functions longer, but there are some duplicated calls and it's the only place `pull` is used. Up to you.

File: src/ostree.h [hunk 2]
```diff
@@ -24,10 +24,12 @@ class OstreeManager : public PackageManagerInterface {
   boost::shared_ptr<OstreeDeployment> getStagedDeployment();
   static boost::shared_ptr<OstreeSysroot> LoadSysroot(const boost::filesystem::path &path);
   static bool addRemote(OstreeRepo *repo, const std::string &url, const KeyManager &keys);
-  static data::InstallOutcome pull(const Config &config, const KeyManager &keys, const std::string &refhash);
+  statuc data::InstallOutcome OstreeManager::find_or_pull(const Config &config, const KeyManager &keys,
```
Comment (patrickvacek) [inline src/ostree.h]: `statuc`?

Comment (patrickvacek) [reply src/ostree.h]: The `OstreeManager::` is redundant, too.

Comment (OYTIS) [reply src/ostree.h]: Sorry, compiled with BUILD_OSTREE=OFF before pushing.

File: src/ostree.cc [hunk 3]
```diff
@@ -55,6 +55,27 @@ data::InstallOutcome OstreeManager::pull(const Config &config, const KeyManager
   return data::InstallOutcome(data::OK, "Pulling ostree image was successful");
 }
 
+data::InstallOutcome OstreeManager::find_or_pull(const Config &config, const KeyManager &keys,
```
Comment (patriotyk) [inline src/ostree.cc]: CamelCase

**Comment by patrickvacek** [comment]
I think this looks fine, but I'm getting an error when trying to update my device. This could be user error. Have you gotten it to work successfully yet?

**Comment by OYTIS** [comment]
Wrong libostree function, sorry.

**Comment by OYTIS** [comment]
Now that should work. Tested on rocko with autoprovisioning.

**Pull request merged**